addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -427,7 +427,8 @@ var NumericField = InputField.extend({
      * @private
      */
     _formatValue: function (value) {
-        if (this.mode === 'edit' && this.nodeOptions.type === 'number') {
+        const doFormat = Boolean(JSON.parse('format' in this.nodeOptions ? this.nodeOptions.format : true));
```

Comment:
I'd do this in init, s.t. it is done only once
this.shouldFormat = Boolean(...); // be careful of 100 char hard limit